app/com/lucidchart/open/cashy/controllers/UploadController.scala

```diff
@@ -148,52 +139,44 @@ class UploadController extends AppController with ExtensionsConfig with UploadFe
   def krakenPreview = AuthAction.authenticatedUser { implicit user =>
     Action(parse.multipartFormData(FileHandler.handleFilePartAsByteArray)) { implicit request =>
 
-      val fileOption = request.body.file("assetFile")
       val formWithData = krakenForm.bindFromRequest()
       try {
-        fileOption match {
-          case None => {
-            throw new Exception("File not found")
-          }
-          case Some(file) => {
-            formWithData.fold(
-              formWithErrors => {
-                throw new Exception(formWithErrors.errors.map(e => e.key + ": " + e.message)mkString("\n"))
-              },
-              data => {
-                // Get the file bytes and content type
-                val (bytes, contentType, filename) = file match {
-                  case FilePart(key, filename, contentType, bytes) => {
-                    (bytes, contentType, filename)
-                  }
-                }
+        formWithData.fold(
+          formWithErrors => {
+            throw new UploadFailedException(formWithErrors.errors.map(e => e.key + ": " + e.message)mkString("\n"))
+          },
+          data => {
 
-                val bucket = data.bucket
-                val width = data.width
-                val height = data.height
+            // Get the asset data
+            val assetData = AssetDataHelper.getData(request.body.file("assetFile"), data.assetURL)
 
-                val extension = getExtension(filename)
+            val bytes = assetData.bytes
```

Comment:
Do we really need to create all these temporary variables, or can we just reference them with dot notation when needed?